### PR TITLE
Go through the docs, correct grammar, etc

### PR DESCRIPTION
**Pull request by asinghvi17**
Sorry for the large PR but this is just the result of me running an AI over the docs.  Not urgent at all, feel free to merge/review or not.

### CODE REVIEW

File: docs/src/cuda.md [hunk 1]
```diff
@@ -50,20 +50,20 @@ CuArray{Float32, 2, CUDA.Mem.DeviceBuffer}
 
 DimensionalData.jl has two GPU-related goals:
 
-1. Work seamlessly with Base julia broadcasts and other operations that already
+1. Work seamlessly with base Julia broadcasts and other operations that already
```
Comment (lazarusA) [inline docs/src/cuda.md:53]:
```suggestion
1. Work seamlessly with `Base` Julia broadcasts and other operations that already
```

File: docs/src/selectors.md [hunk 2]
```diff
@@ -118,7 +115,7 @@ A[X=Where(>=(1.5)), Y=Where(x -> x in (:a, :c))]
 
 == Not
 
-`Not(x)` get all indices _not_ selected by `x`, which can be another selector.
+The [`Not(x)`](@ref) selector gets all indices _not_ selected by `x`, which can be another selector.
```
Comment (lazarusA) [inline docs/src/selectors.md:118]: this one is failing in CI, not sure why....

File: docs/src/stacks.md [hunk 3]
```diff
@@ -12,8 +12,8 @@ x, y = X(1.0:10.0), Y(5.0:10.0)
 st = DimStack((a=rand(x, y), b=rand(x, y), c=rand(y), d=rand(x)))
 ````
 
-The behaviour of a `DimStack` is at times like a `NamedTuple` of
-`DimArray` and, others an `AbstractArray` of `NamedTuple`.
+The behavior of a `DimStack` is at times like a `NamedTuple` of
```
Comment (lazarusA) [inline docs/src/stacks.md:15]:
```suggestion
The behaviour of a `DimStack` is at times like a `NamedTuple` of
```

**Review by lazarusA — CHANGES_REQUESTED** [review]
Looks good to me, we should merge after fixing the ```[`Not(x)`](@ref)``` fail in CI.

File: docs/src/selectors.md (hunk 2 above)
Comment (rafaqz) [reply docs/src/selectors.md:118]: We don't own it so we can't use ref

Comment (rafaqz) [inline docs/src/selectors.md:118]:
```suggestion
The `Not(x)` selector gets all indices _not_ selected by `x`, which can be another selector.
```

File: docs/src/tables.md [hunk 4]
```diff
@@ -1,18 +1,10 @@
...
 
-DimensionalData.jl implements the Tables.jl interface for
-`AbstractDimArray` and `AbstractDimStack`. `DimStack` layers
-are unrolled so they are all the same size, and dimensions
-loop to match the length of the largest layer.
+DimensionalData.jl implements the Tables.jl interface for `AbstractDimArray` and `AbstractDimStack`. `DimStack` layers are unrolled so they are all the same size, and dimensions loop to match the length of the largest layer.
 
-Columns are given the [`name`](@ref) or the array or the stack layer key.
-`Dimension` columns use the `Symbol` version (the result of `DD.name(dimension)`).
+Columns are given the [`name`](@ref) of the array or the stack layer key, or the result of `DD.name(dimension)` for `Dimension` columns.
```
Comment (rafaqz) [inline docs/src/tables.md:7]:
```suggestion
Columns are given the [`name`](@ref) of the array or stack layer, and the result of `DD.name(dimension)` for `Dimension` columns.
```

**Review by rafaqz — APPROVED** [review]

**Pull request merged**